manager/kubernetes.go

```diff
@@ -34,9 +34,17 @@ func (m *VolumeManager) PVCreate(name, pvName, fsType, secretNamespace, secretNa
 		pvName = v.Name
 	}
 
-	storageClassName, err := m.ds.GetSettingValueExisted(types.SettingNameDefaultLonghornStaticStorageClass)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get longhorn static storage class name for PV %v creation: %v", pvName, err)
+	if storageClassName == "" {
+		storageClassName, err = m.ds.GetSettingValueExisted(types.SettingNameDefaultLonghornStaticStorageClass)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get longhorn static storage class name for PV %v creation: %v", pvName, err)
+		}
+		if backupVolumeName, isExist := v.Labels[types.LonghornLabelBackupVolume]; isExist && backupVolumeName != "" {
+			backupVolume, _ := m.ds.GetBackupVolumeRO(backupVolumeName)
+			if backupVolume != nil {
+				storageClassName = backupVolume.Status.StorageClassName
+			}
+		}
 	}
```

Comment:
Can do it reverse, also make it clear to understand how storaglcass is determined. 

```go
if storageClassName == "" {
    if backupVolumeName := v.Labels[types.LonghornLabelBackupVolume]; backupVolumeName != "" {
        backupVolume, _ := m.ds.GetBackupVolumeRO(backupVolumeName)
        if backupVolume != nil {
            storageClassName = backupVolume.Status.StorageClassName
        }
    }
}

if storageClassName == "" {
    storageClassName, err = m.ds.GetSettingValueExisted(types.SettingNameDefaultLonghornStaticStorageClass)
    if err != nil {
    return nil, fmt.Errorf("failed to get longhorn default static storage class name for PV %v creation: %v", pvName, err)
    }
}

```

Reply:
Thanks, this looks better!